### ISSUE TITLE
Dedent content in Markdown compoent to make user code cleaner

### ISSUE THREAD

**Issue by FHU-yezi** [pynecone-io/pynecone#158]
When I type this code:

```python
def index():
    return pc.vstack(
        pc.markdown(
            """
            This is a test line
                and test again but with indent
                    third, more indent
            finally, remove all indents
            """
        ),
        margin_top="2em"
    )
```

It seems like this:

![](https://user-images.githubusercontent.com/50828953/209023867-dc5d7250-d454-4e03-aa8d-7117bd20c962.png)

It rendered as a code block, because in Markdown syntax, mulitline text with indent is same as code block.

So I have to modify my code like this:

```python
def index():
    return pc.vstack(
        pc.markdown(
"""
This is a test line
    and test again but with indent
        third, more indent
finally, remove all indents
"""
        ),
        margin_top="2em"
    )
```

And... Why my text rendered in one single line?

![](https://user-images.githubusercontent.com/50828953/209024326-e0447177-bd37-4f7d-9bed-aacbf799a40f.png)

Maybe add some `\n` will fix this problem, let's try:

```python
def index():
    return pc.vstack(
        pc.markdown(
"""
This is a test line\n
    and test again but with indent\n
        third, more indent\n
finally, remove all indents\n
"""
        ),
        margin_top="2em"
    )
```

![](https://user-images.githubusercontent.com/50828953/209024396-b7a9bac5-0b0f-4010-b4a2-4f336a1c62be.png)

It works, but why we have different behavior between indent and no-indent?

Why I should dedent my code when a mulitline block start?

I tried to fix it by myself, modified `component/typography/markdown.py`, line 34, but it doesn't work.

Python standard library `textwrap` has a function named `dedent` can do this thing.

**Comment by picklelo** [comment]
@FHU-yezi Markdown requires two enters in order to get a line-break (Just going to the new line continues the text in the same line). This code should do the breaks as you want:

```
def index():
    return pc.vstack(
        pc.markdown(
"""
This is a test line

and test again but with indent

third, more indent

finally, remove all indents
"""
        ),
        margin_top="2em"
    )
   ```

You may have to use some HTML character like `&nbsp` to actually indent the line.

I don't think there's anything we should do to automatically indent/dedent the markdown string.

**Comment by FHU-yezi** [comment]
Yes I already use Markdown to write my articles. As you said we need to press enter for two times to start a new line, and use `&nbsp` to indent content.

I means if we can automaticlly dedent the `src` string, user can indent there text block to the same level which other code in.

So I needn't to write this:

```
def index():
    return pc.markdown(
'''
Some text.
'''
    )
```

I can write this:

```
def index():
    return pc.markdown(
        '''
        Some text.
        '''
    )
```

This is prettier.

**Comment by FHU-yezi** [comment]
Yes I already use Markdown to write my articles. As you said we need to press enter for two times to start a new line, and use `&nbsp` to indent content.

I means if we can automaticlly dedent the `src` string, user can indent there text block to the same level which other code in.

So I needn't to write this:

```python
def index():
    return pc.markdown(
'''
Some text.
'''
    )
```

I can write this:

```python
def index():
    return pc.markdown(
        '''
        Some text.
        '''
    )
```

This is prettier.

**Comment by picklelo** [comment]
Agreed - actually there is a bug in the Markdown component. Because the correct way to do it should be to set the `src` prop like
```
pc.markdown(
    src="""
    This is a test line
    """
),
```
but this seems to be not working very well for complicated strings. We will fix this and add the automatic dedenting.
